File: 1.0/src/c/cz/mg/vulkan/PFN_vkEnumeratePhysicalDevices.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkEnumeratePhysicalDevices_callNative(JNIEnv* env, jclass jc, jlong address, jlong instance, jlong pPhysicalDeviceCount, jlong pPhysicalDevices, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkEnumeratePhysicalDevices f = (PFN_vkEnumeratePhysicalDevices)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkInstance*)jniLongToPointer(instance)),
        ((uint32_t*)jniLongToPointer(pPhysicalDeviceCount)),
        ((VkPhysicalDevice*)jniLongToPointer(pPhysicalDevices))
    );
}


jint Java_cz_mg_vulkan_PFNvkEnumeratePhysicalDevices_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong instance, jlong pPhysicalDeviceCount, jlong pPhysicalDevices){
    (void)env;
    (void)jc;
    PFN_vkEnumeratePhysicalDevices f = (PFN_vkEnumeratePhysicalDevices)jniLongToFunctionPointer(address);
    return f(
        *((VkInstance*)jniLongToPointer(instance)),
        ((uint32_t*)jniLongToPointer(pPhysicalDeviceCount)),
        ((VkPhysicalDevice*)jniLongToPointer(pPhysicalDevices))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkEnumeratePhysicalDevices.java
package cz.mg.vulkan;

public class PFNvkEnumeratePhysicalDevices extends VkFunctionPointer {
    public PFNvkEnumeratePhysicalDevices() {
    }

    protected PFNvkEnumeratePhysicalDevices(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkEnumeratePhysicalDevices(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkEnumeratePhysicalDevices(long value) {
        setValue(value);
    }

    public PFNvkEnumeratePhysicalDevices(VkInstance instance) {
        super(instance, new VkString("vkEnumeratePhysicalDevices"));
    }

    public void call(VkInstance instance, VkUInt32 pPhysicalDeviceCount, VkPhysicalDevice pPhysicalDevices, VkResult rval){
        callNative(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getNullAddressNative(), pPhysicalDeviceCount != null ? pPhysicalDeviceCount.getVkAddress() : VkPointer.NULL, pPhysicalDevices != null ? pPhysicalDevices.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long instance, long pPhysicalDeviceCount, long pPhysicalDevices, long rval);


    public int call(VkInstance instance, VkUInt32 pPhysicalDeviceCount, VkPhysicalDevice pPhysicalDevices){
        return callSimplifiedNative(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getNullAddressNative(), pPhysicalDeviceCount != null ? pPhysicalDeviceCount.getVkAddress() : VkPointer.NULL, pPhysicalDevices != null ? pPhysicalDevices.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long instance, long pPhysicalDeviceCount, long pPhysicalDevices);

}
